tests/kokkos/workspace_tests.cpp

```diff
@@ -152,27 +152,33 @@ static void unittest_workspace()
     team.team_barrier();
 
     Kokkos::Array<Unmanaged<view_1d<int> >, num_ws> wssub;
+    Unmanaged<view_2d<int> wsmacro;
+    const auto space = ws.take_macro_block("space", num_ws);
 
     // Main test. Test different means of taking and release spaces.
-    for (int r = 0; r < 8; ++r) {
-      if (r % 4 == 0) {
+    for (int r = 0; r < 10; ++r) {
+      if (r % 5 == 0) {
```

Comment:
Maybe this is just me, but...this switch on `r % 5` it's' a bit hard to follow. Since the code shared by all code paths is minimal (that I can tell), wouldn't be easier to just get rid of the outer for?

Reply:
Yeah, I'm not sure why it was written this way. Maybe it is so you have to take and release and then take again?

Reply:
The idea is you cycle through 4 (now 5) different actions twice.